### PR TITLE
Fix filename sanitization for downloaded files (do not replace periods, do not change case, and do not leave Unicode characters decomposed)

### PR DESCRIPTION
**Pull request by ViciousEagle03**
This PR address the issue #7450

### CODE REVIEW

File: sunpy/util/tests/test_net.py [hunk 1]
```diff
@@ -21,6 +21,7 @@ def test_slugify():
     assert sunpy.util.net.slugify("äb c", "b_c")
     assert sunpy.util.net.slugify("file.greg.fits") == "file_greg.fits"
     assert sunpy.util.net.slugify("file.greg.fits", "x") == "filexgreg.fits"
+    assert sunpy.util.net.slugify("file.name.fits.hdu") == "file_name.fits.hdu"
```
Comment (nabobalis) [inline sunpy/util/tests/test_net.py:24]: Can you add the examples from the issue so we can double check?

**Comment by nabobalis** [comment]
Can you also add a changelog please.

File: sunpy/util/tests/test_net.py (hunk 1 above)
Comment (ViciousEagle03) [reply sunpy/util/tests/test_net.py:24]: Sure i will add the example,

**Comment by nabobalis** [comment]
Actually in line with https://github.com/sunpy/sunpy/issues/7450#issuecomment-1951436626, I think we should leave periods alone.

**Comment by ViciousEagle03** [comment]
> Actually in line with [#7450 (comment)](https://github.com/sunpy/sunpy/issues/7450#issuecomment-1951436626), his point makes a lot of sense. I think we should leave the periods alone.
So should I remove my fix in `net.py` and remove the periods from the list that `slugify()` function replaces?

**Comment by ViciousEagle03** [comment]
So should I remove my fix in `net.py` and remove the periods from the list that `slugify()` function replaces?

**Comment by nabobalis** [comment]
> So should I remove my fix in `net.py` and remove the periods from the list that `slugify()` function replaces?

Yes please.

**Comment by samaloney** [comment]
Thanks for the PR this addresses the specific issue and look good to me. 

It would be good to consider if could address the the root cause see the discussion on #7450

**Comment by ViciousEagle03** [comment]
Sure, I will look into the root cause

**Comment by ayshih** [comment]
I've directly committed to this PR to re-implement `slugify()` in a more straightforward way

**Comment by ayshih** [comment]
For the Unicode fans out there, `slugify()` has been normalizing to NFKD form (decomposed by compatibility), but I feel it should normalize to NFKC form (decomposed by compatibility, then recomposed by canonical equivalence).  It's weird to me that it characters decomposed.  For example, "ä" becomes two Unicode characters ("a" + U+0308) instead of being left intact.  Any thoughts?

I'll note that there is currently a test line that looks intended to test the decomposition of "ä", but that line is bugged, so the output isn't tested.

**Comment by nabobalis** [comment]
I would be in favour of NFKC

**Comment by nabobalis** [comment]
Online Test failures look to be real.

One is a filename difference, and the other seems to be more results returned from the VSOClient?

**Comment by ViciousEagle03** [comment]
Do you have any tasks for at the moment
Apologies for not being active last week I was occupied with my mid-terms

**Comment by nabobalis** [comment]
> Do you have any tasks for me at the moment . Apologies for not being active last week I was occupied with my mid-terms

I think the online tests need updating but that is it.

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks for the PR @ViciousEagle03 and @ayshih

**Comment by nabobalis** [comment]
I decided not to backport this and add a whatsnew as I don't want to break previous behaviour for filenames for the released versions.

**Pull request merged**